sbnana/CAFAna/XSec/Flux.cxx

```diff
@@ -90,6 +93,70 @@ namespace ana
     return ret;
   }
 
+  //----------------------------------------------------------------------
+  void FluxTimesNuclei::SaveTo(TDirectory* dir, const std::string& name) const
+  {
+    TDirectory* tmp = gDirectory;
+
+    dir = dir->mkdir(name.c_str()); // switch to subdir
+    dir->cd();
+
+    TObjString("FluxTimesNuclei").Write("type");
+
+    TVectorD pdg(1);
+    pdg[0] = fPdg;
+    pdg.Write("pdg");
+
+    Spectrum::SaveTo(dir, "spectrum");
+
+    dir->Write();
+    delete dir;
+
+    tmp->cd();
+  }
+
+  //----------------------------------------------------------------------
+  std::unique_ptr<FluxTimesNuclei> FluxTimesNuclei::LoadFrom(TDirectory* topdir, const std::string& name)
+  {
+    std::unique_ptr<TDirectory> dir(topdir->GetDirectory(name.c_str())); // switch to subdir
+    assert(dir);
+
+    DontAddDirectory guard;
+
+    std::unique_ptr<TObjString> tag((TObjString*)dir->Get("type"));
+    assert(tag);
+    assert(tag->GetString() == "FluxTimesNuclei");
+
+    // std::unique_ptr<int> pdg(new int(1));
+    std::unique_ptr<TVectorD> pdg((TVectorD*)dir->Get("pdg"));
+    assert(pdg && pdg->GetNrows()==1);
+
+    return std::unique_ptr<FluxTimesNuclei>(new FluxTimesNuclei(Spectrum::LoadFrom(dir.get(), "spectrum"), (*pdg)[0]));
+  }
+
+  //----------------------------------------------------------------------
+  FluxTimesNuclei::FluxTimesNuclei(const std::unique_ptr<Spectrum> spec, const int pdg)
+    : Spectrum(*spec), fPdg(pdg)
+  {
+  }
+
+  //----------------------------------------------------------------------
+  Spectrum FluxTimesNuclei::MakeTotalFlux(const HistAxis& ax) const
+  {
+    const unsigned int nbins = ax.GetBins1D().NBins()+2;
+
+    // Hist h = Hist::Zero(nbins);
+
+    const double thisIntegral(this->Integral(fPOT));
+    // for(unsigned int bin = 0; bin < nbins; bin++)
+    //   h.Fill(bin, thisIntegral);
+
+    Eigen::ArrayXd data(nbins);
+    data.setConstant(thisIntegral);
+
+    return Spectrum(data, LabelsAndBins(ax), fPOT, fLivetime);
```

Comment:
I don't think this cast is necessary
```suggestion
    return Spectrum(data, ax, fPOT, fLivetime);
```
Though, in fact, this function should take a LabelsAndBins, not HistAxis, in the first place.